core/unit_test/TestSharedHostPinnedSpace.cpp

```diff
@@ -0,0 +1,132 @@
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+template <typename ViewType>
+struct Increment {
+  ViewType view_;
+
+  template <typename ExecutionSpace>
+  explicit Increment(ExecutionSpace, ViewType view) : view_(view) {
+    Kokkos::fence();
+    Kokkos::parallel_for(
+        "increment",
+        Kokkos::RangePolicy<ExecutionSpace, Kokkos::IndexType<size_t>>{
+            0, view_.size()},
+        *this);
+    Kokkos::fence();
+  }
+
+  KOKKOS_FUNCTION
+  void operator()(const size_t idx) const { ++view_(idx); }
+};
+
+template <typename ViewType>
+struct CheckResult {
+  ViewType view_;
+  const int targetVal_;
+  unsigned error = 0;
+
+  template <typename ExecutionSpace>
+  CheckResult(ExecutionSpace, ViewType view, int targetVal)
+      : view_(view), targetVal_(targetVal) {
+    Kokkos::parallel_reduce(
+        "check",
+        Kokkos::RangePolicy<ExecutionSpace, Kokkos::IndexType<size_t>>{
+            0, view_.size()},
+        *this, Kokkos::Sum<unsigned>(error));
+    Kokkos::fence();
+  }
+
+  KOKKOS_FUNCTION
+  void operator()(const size_t idx, unsigned& error_val) const {
+    if (view_(idx) != targetVal_) ++error_val;
+  }
+};
+
+TEST(defaultdevicetype, shared_host_pinned_space) {
+  ASSERT_TRUE(KOKKOS_HAS_SHARED_HOST_PINNED_SPACE);
+  ASSERT_TRUE(Kokkos::has_shared_host_pinned_space);
+
+  if constexpr (std::is_same_v<Kokkos::DefaultExecutionSpace,
+                               Kokkos::DefaultHostExecutionSpace>)
+    GTEST_SKIP() << "Skipping as host and device are the same space";
+
+  const unsigned int numDeviceHostCycles = 3;
+  size_t numInts                         = 1024;
+
+  // ALLOCATION
+  Kokkos::View<int*, Kokkos::SharedHostPinnedSpace> sharedData("sharedData",
+                                                               numInts);
+  // MAIN LOOP
+  unsigned incrementCount = 0;
+
+  for (unsigned i = 0; i < numDeviceHostCycles; ++i) {
+    // INCREMENT DEVICE
+    Increment incrementDevice(Kokkos::DefaultExecutionSpace{}, sharedData);
+    ++incrementCount;
+    // CHECK RESULTS HOST
+    CheckResult checkHost(Kokkos::DefaultHostExecutionSpace{}, sharedData,
+                          incrementCount);
+    ASSERT_TRUE(checkHost.error == 0)
```

Comment:
```suggestion
    ASSERT_EQ(checkHost.error, 0)
```
Same comment below.  That is the only one that you must fix.  Others I will leave up to you.

Reply:
done in [0fac2a](https://github.com/kokkos/kokkos/pull/5405/commits/0fac2a538480266303a158ffec5cb8c52f04962b)